flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SplitAssignmentTracker.java

```diff
@@ -71,7 +68,8 @@ public void snapshotState(
         // Include the uncheckpointed assignments to the snapshot.
         assignmentsByCheckpointId.put(checkpointId, uncheckpointedAssignments);
         uncheckpointedAssignments = new HashMap<>();
-        writeAssignmentsByCheckpointId(assignmentsByCheckpointId, splitSerializer, out);
```

Comment:
I guess this is a breaking change. Old version will fails in restoring (EOFException ?). How about a forward compatible change ? It should similar to how FLINK-21452 do in [pr-14994](https://github.com/apache/flink/pull/14994/files#diff-9144631bff9092280da424a52db7a60d2e0f47e82b0c55ccbdc93ac29614d75cL291). I known Flink does not support downgrade, but I think it is might be a good for this minor change to not breaking anything.

Reply:
Thanks @kezhuw ! That's a good catch. I think writing an empty map here will keep the forward compatibility. I'll update my code later.

Reply:
Fixed in 50242f99a935c9d832dbc33e5c51ad3d2f81a05e